modules/abstract-utxo/src/parseOutput.ts

```diff
@@ -222,6 +222,29 @@ export async function parseOutput({
   let currentAddressType: string | undefined = undefined;
   const RECIPIENT_THRESHOLD = 1000;
   try {
+    // In the case of PSBTs, we can already determine the internal/external status of the output addresses
+    // based on the derivation information being included in the PSBT. We can short circuit GET v2.wallet.address
+    // and save on network requests. Since we have the derivation information already, we can still verify the address
+    if (currentOutput.external !== undefined) {
+      // In the case that we have a custom change wallet, we need to verify the address against the custom change keys
+      // and not the wallet keys. This check is done in the handleVerifyAddressError function if this error is thrown.
+      if (customChange !== undefined) {
+        throw new UnexpectedAddressError('`address validation failure');
+      }
+      // If it is an internal address, we can skip the network request and just verify the address locally with the
+      // derivation information we have. Otherwise, if the address is external, which is the only remaining case, we
+      // can just return the current output as is without contacting the server.
+      if (isWalletOutput(currentOutput)) {
+        await coin.isWalletAddress({
```

Comment:
The name suggests that this returns a boolean - if that's not the case please add a comment. Otherwise check the result.